Lib/dataclasses.py

```diff
@@ -228,6 +228,9 @@ class InitVar:
     def __init__(self, type):
         self.type = type
 
+    def __call__(self, *args, **kwargs):  # required by typing.get_type_hints, see bpo-44799
```

Comment:
How does this solve the problem at hand?

I'm unlikely to accept any changes to dataclasses and typing until PEP 649 is ruled on.

Reply:
Exception raised by `_type_check`. In case if object is callable this object won't raise exception.

https://github.com/python/cpython/blob/e06ae75e16dbf46b6626b6a41b62391ea1fdb319/Lib/typing.py#L180-L181

I found that it was the easiest way to solve this issue. If you think it's a bad solution I am open to fix it in another way.

Reply:
It just seems like it's papering over some actual problem.

In any event, I'm waiting on PEP 649, which if accepted makes this whole point moot.

Reply:
Makes sense, will wait on a update regarding PEP 649.